### ISSUE TITLE
[Checker] Unnecessary encoding specification (PEP 263, 3120)

### ISSUE THREAD

**Issue by leezu** [pylint-dev/pylint#3106]
### Is your feature request related to a problem? Please describe
In Python 3, UTF-8 is the default source encoding.

Statements such as `# coding=utf-8` in the first two lines of a program are not necessary and should be deleted. In general, the use of any other encoding but utf-8 is probably to be discouraged.

### Describe the solution you'd like
Pylint should check that none of the first two lines of a python file matches
`^[ \t\f]*#.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)` [1].

### Additional context
[1]: Regex taken from https://www.python.org/dev/peps/pep-0263/

https://www.python.org/dev/peps/pep-3120 specifies the change in default encoding.

**Comment by PCManticore** [comment]
Sounds good to me.

**Comment by valerybriz** [comment]
Hello, I would like to help with this issue :)

**Comment by dutchgecko** [comment]
Additionally, warnings `W9903` and `W9905` should not be run by default on Python 3 imo.

**Comment by Pierre-Sassoulas** [comment]
@valerybriz do you still want to help ? I can assign you to the issue if you want :)

**Comment by clavedeluna** [comment]
Related to checker `bad-file-encoding`, but in this case we're saying of regex matches for utf-8, pylint should instead output a new message, something like `useless-encoding` or `unnecessary-file-encoding`.

**Comment by clavedeluna** [comment]
> `^[ \t\f]*#.*?coding[:=][ \t]*([-_.a-zA-Z0-9]+)` [1].

I don't think this is the regex we want, given it would also match `# coding: iso-8859-1`.

**Comment by Pierre-Sassoulas** [comment]
> I don't think this is the regex we want, given it would also match # coding: iso-8859-1.

If we want to push the use of utf8 for everything then it make sense to prevent any encoding specification at all. I think utf32 is better if you have a lot of non ascii characters but mainly pushing hard for utf8 make sense.
